src/main/java/io/confluent/connect/hdfs/partitioner/TimeBasedPartitioner.java

```diff
@@ -86,14 +88,23 @@ public void configure(Map<String, Object> config) {
     String hiveIntString = (String) config.get(HdfsSinkConnectorConfig.HIVE_INTEGRATION_CONFIG);
     boolean hiveIntegration = hiveIntString != null && hiveIntString.toLowerCase().equals("true");
 
+    timeFieldName = (String) config.get(HdfsSinkConnectorConfig.PARTITION_TIME_FIELD_NAME_CONFIG);
+
     Locale locale = new Locale(localeString);
     DateTimeZone timeZone = DateTimeZone.forID(timeZoneString);
     init(partitionDurationMs, pathFormat, locale, timeZone, hiveIntegration);
   }
 
   @Override
   public String encodePartition(SinkRecord sinkRecord) {
-    long timestamp = System.currentTimeMillis();
+    long timestamp;
+    if (timeFieldName.equals("")) {
+      timestamp = System.currentTimeMillis();
+    } else {
+      Struct struct = (Struct) sinkRecord.value();
+      timestamp = (long) struct.get(timeFieldName);
```

Comment:
There are also logical types (e.g. `Timestamp`) that you might want to support here.